src/app/page.tsx

```diff
@@ -1,41 +1,37 @@
-import {
-	EstimatedPackage,
-	PackageOwnership,
-	tideliftMeUp,
-} from "tidelift-me-up";
-
 import { Footer } from "~/components/Footer";
 import { MainArea } from "~/components/MainArea";
 import { OptionsForm } from "~/components/OptionsForm";
 import { ResultDisplay } from "~/components/ResultDisplay";
 import { ScrollButton } from "~/components/ScrollButton";
+import { SearchParamsType, fetchData } from "~/utils/fetchData";
 
+import { metadata as defaultMetadata } from "./layout";
 import styles from "./page.module.css";
 
 export interface HomeProps {
-	searchParams: Record<string, unknown>;
+	searchParams: SearchParamsType;
 }
 
-export default async function Home({ searchParams }: HomeProps) {
-	const options = {
-		ownership: undefinedIfEmpty(
-			[
-				searchParams["author"] === "on" && "author",
-				searchParams["maintainer"] === "on" && "maintainer",
-				searchParams["publisher"] === "on" && "publisher",
-			].filter(Boolean) as PackageOwnership[],
-		),
-		since: (searchParams["since"] || undefined) as string | undefined,
-		username: searchParams.username as string,
-	};
-	let result: Error | EstimatedPackage[] | undefined;
+export async function generateMetadata({ searchParams }: HomeProps) {
+	const { options, result } = await fetchData(searchParams);
+	const username = options.username || "";
+	const packageCount = Array.isArray(result) ? result.length : 0;
```

Comment:
[Non-Actionable] Noting from QA: there's no way in the tidelift-me-up package to know that the reason for no packages is that the user didn't exist. Tracked here: https://github.com/JoshuaKGoldberg/tidelift-me-up/issues/17. So, explicitly saying I don't think we need to block the PR on giving a specific description for that. IF you really want to get that in first I'm ok waiting, but otherwise by default I'd just suggest we go with a followup issue. What do you think?

> ```plaintext
> Could not find {username} on npm.
> ```

Reply:
I agree with not blocking this PR and instead creating a follow up issue to address this bug. I went ahead and created an [issue](https://github.com/JoshuaKGoldberg/tidelift-me-up-site/issues/844) so this can be fixed when the tidelift-me-up package is updated.

Reply:
Thanks!